integrated_channels/api/v1/mixins.py

```diff
@@ -0,0 +1,56 @@
+"""
+Mixins for Integrated Channel API Viewsets
+"""
+from uuid import UUID
+
+from edx_rbac.mixins import PermissionRequiredForListingMixin
+from rest_framework.exceptions import ParseError
+
+from enterprise.constants import ENTERPRISE_ADMIN_ROLE
+from enterprise.models import SystemWideEnterpriseUserRoleAssignment
+
+
+class PermissionRequiredForIntegratedChannelMixin(PermissionRequiredForListingMixin):
+    """
+    `configuration_model` (class) - The RoleAssignmentClass against which DB-defined access is checked.
+    """
+    # fields that control permissions for 'list' actions
+    list_lookup_field = 'enterprise_customer'
+    allowed_roles = [ENTERPRISE_ADMIN_ROLE, ]
+    role_assignment_class = SystemWideEnterpriseUserRoleAssignment
+
+    @property
+    def requested_enterprise_uuid(self):
+        enterprise_customer_uuid = self.request.query_params.get('enterprise_customer')
+        if not enterprise_customer_uuid:
+            return None
+        try:
+            return UUID(enterprise_customer_uuid)
+        except ValueError:
+            raise ParseError('{} is not a valid uuid.'.format(enterprise_customer_uuid))
+
+    def get_permission_object(self):
+        """
+        Used for "retrieve" actions. Determines the context (enterprise UUID) to check
+        against for role-based permissions.
+        """
+        if self.requested_enterprise_uuid:
+            return self.requested_enterprise_uuid
+        enterprise_customer_uuid = self.request.data.get('enterprise_customer_uuid')
```

Comment:
this may be dumb, but is there a reason we'd never go through the `requested_enterprise_uuid` object? Ie what I'm seeing here is us looking to see if we have the UUID defined in this property already and if not, we look at the query params and take the same steps as that property, but adhoc/separately.

Reply:
Missed the refactor on that, so yes it was me being dumb!